### PR TITLE
docs: `arXiv` references page

### PR DESCRIPTION
**Pull request by leo-gan**
Since the LangChain based on many research papers, the LC documentation has several references to the arXiv papers. It would be beneficial to create a single page with all referenced papers.
PR:
1. Developed code to search the arXiv references in the LangChain Documentation and the LangChain code base. Those references are included in a newly generated documentation page. 
2. Page is linked to the Docs menu.

Controversial:
1. The `arxiv_references` page is automatically generated. But this generation now started only manually. It is not included in the doc generation scripts. The reason for this is simple. I don't want to mangle into the current documentation refactoring. If you think, we need to regenerate this page in each build, let me know. Note: This script has a dependency on the `arxiv` package.
2. The link for this page in the menu is not obvious.

### CODE REVIEW

**Comment by eyurtsev** [comment]
https://langchain-git-fork-leo-gan-docs-arxiv-references-langchain.vercel.app/docs/additional_resources/arxiv_references

**Comment by eyurtsev** [comment]
Looks good I think. @leo-gan should it be titled arXiv or literature? in case folks don't know about the arxiv

**Comment by leo-gan** [comment]
> Looks good I think. @leo-gan should it be titled arXiv or literature? in case folks don't know about the arxiv

IMHO, if folks don't know about the arXiv, they don't need these references. I named the page the "arXiv" in the menu. So, I changed the title as you suggested, it would be in the line with menu.

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
Updating to new master, and will merge as long as unit tests pass!

File: docs/scripts/arxiv_references.py [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+    Search for the arXiv references in the documentation pages.
+    Note: It finds only the first arXiv reference in a line.
+
+    Args:
+        docs_dir: Path to the documentation root folder.
+    Returns:
+        dict: Dictionary with arxiv_id as key and set of file names as value.
+    """
+    arxiv_url_pattern = re.compile(ARXIV_ID_PATTERN)
+    exclude_strings = {"file_path", "metadata", "link", "loader", "PyPDFLoader"}
```
[the remark this answers is not in the document]
Comment (leo-gan) [reply docs/scripts/arxiv_references.py:51]: Do you mean the exclude_strings? It holds the empirical values. These arXive references are the part of the example values not the references in the text. For example, this line
`  'link': 'https://arxiv.org/abs/2206.12839',` - it is a code line in one of the notebooks.

**Pull request merged**